### PR TITLE
Chapter 2

### PR DESCRIPTION
**Pull request by hitesh-cap**

### CODE REVIEW

File: firsttestcase.test.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        axios.mockImplementation(() => 
+            Promise.resolve({data:{ joke: "Hello World!", error: false}})
+        );
+        const res = await request(app).get("/test/securedRandomJoke").query({ isSuperUser: 1 });
+        expect(res.text).toBe("Hello World!");
+    });
+    it("sucessfull API request secured", async()=>{
+        const res= await request(app).get("/test/securedRandomJoke");
+        expect(res.statusCode).toBe(200);
+    })
```
Comment (Vishakhavel) [inline firsttestcase.test.js:48]: nice job!

**Comment by Vishakhavel** [comment]
looks good. time to merge.

File: src/mapper/calculator.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      });
+    }
+    case CONSTANTS.CALCULATOR_OPERATIONS.MULTIPLICATION: {
+      return values.reduce((sum, value) => {
+        return sum * value
+      });
+    }
+    case CONSTANTS.CALCULATOR_OPERATIONS.DIVISION: {
+      return values[1] / values[0];
+    }
```
Comment (preetham-b-capillary) [inline src/mapper/calculator.js:23]: divided by errior

**Review by preetham-b-capillary — CHANGES_REQUESTED** [review]

File: src/mapper/calculator.js (hunk 2 above)
Comment (preetham-b-capillary) [reply src/mapper/calculator.js:23]: done

**Review by preetham-b-capillary — APPROVED** [review]

**Review by preetham-b-capillary — CHANGES_REQUESTED** [review]
please do the above changes about divided by error

File: firsttestcase.test.js [hunk 3]
```diff
@@ -1 +1,46 @@
...
+    it("subtraction check",()=>{
+        expect(calculator('subtraction',2,3)).toBe(-1);
+    })
+    it("multiplication check",()=>{
+        expect(calculator('multiplication',1,2,3)).toBe(6);
+    })
+    it("division check",()=>{
+        expect(calculator('division',1,2)).toBe(2);
+    })
```
Comment (milind30082000) [inline firsttestcase.test.js:20]: Add a case of division by zero also.

**Review by milind30082000 — CHANGES_REQUESTED** [review]

File: firsttestcase.test.js (hunk 3 above)
Comment (hitesh-cap) [reply firsttestcase.test.js:20]: done

**Review by milind30082000 — APPROVED** [review]

**Pull request closed without merging**